crates/resolver/src/async_resolver.rs

```diff
@@ -215,44 +242,81 @@ impl AsyncResolver {
     /// # Returns
     ///
     //  A future for the returned Lookup RData
-    pub fn lookup<N: IntoName>(&self, name: N, record_type: RecordType) -> BackgroundLookup {
+    pub fn lookup<N: IntoName>(
+        &self,
+        name: N,
+        record_type: RecordType,
+        options: DnsRequestOptions,
+    ) -> impl Future<Output = Result<Lookup, ResolveError>> + Send + Unpin + 'static {
```

Comment:
why not make this a async fn? Not like you can really name this right now so you'd have to box it anyways which would make it `Unpin`.

Reply:
So, the experience I had here is that will prevent borrowed `name` parameters that are not `'static`, and that felt too restrictive. Was I doing something wrong?

Reply:
I don't think if you call into_name first it should just borrow it for that first block and drop it. It would be good to test.

The other thing is that I believe also this client should be cheap to clone, just like a channel tx. So it would probably make sense to just make this an async fn so it feels natural to nest it and call await on it. I'd atleast push people in the right direction of writing proper async/await code.

Reply:
Ok, let me play with that.

Reply:
Ok, I've adopted your recommendation where I could. There's one use-case where having `Unpin` is important (maybe that could be removed, but we can do that later)